### PR TITLE
Refactor bauble methods into abstract class

### PR DESCRIPTION
**Pull request by Caedis**

### CODE REVIEW

**Review by Alastors — CHANGES_REQUESTED** [review]
1. What is the point of doing this when it's going to break compat
And 
~~2. Why isn't IBaubleExpanded named IBauble~~ nvm on this part

**Comment by Alastors** [comment]
I'm against randomly making this use an abstract unnecessarily when historically all of baubles uses interfaces and we need to maintain compat with that

**Comment by Caedis** [comment]
No compat is broken. This is mainly for use of standalone bauble items like the to be readded Mining Ring.  

Other items can still implement IBauble or IBaubleExpanded on their own

**Review by mitchej123 — APPROVED** [review]

File: src/main/java/baubles/common/BaubleItemBase.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        setCreativeTab(CreativeTabs.tabTools);
+        this.setMaxStackSize(1);
+    }
+
+    @Override
+    public ItemStack onItemRightClick(ItemStack itemStackIn, World worldIn, EntityPlayer player) {
+        return BaubleItemHelper.onBaubleRightClick(itemStackIn, worldIn, player);
+    }
+
+    @SideOnly(Side.CLIENT)
```
Comment (glowredman) [inline src/main/java/baubles/common/BaubleItemBase.java:30]: `@SideOnly(Side.CLIENT)` is not needed here

File: src/main/java/baubles/common/ItemDebugger.java [hunk 2]
```diff
@@ -4,31 +4,26 @@
...
-import net.minecraft.entity.EntityLivingBase;
-import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.IIcon;
-import net.minecraft.world.World;
 
-public class ItemDebugger extends Item implements IBaubleExpanded {
+public class ItemDebugger extends BaubleItemBase {
+    @SideOnly(Side.CLIENT)
```
Comment (glowredman) [inline src/main/java/baubles/common/ItemDebugger.java:16]: `@SideOnly(Side.CLIENT)` is not needed here

File: src/main/java/baubles/common/ItemDebugger.java [hunk 3]
```diff
@@ -38,11 +33,11 @@ public void registerIcons(IIconRegister ir) {
 		}
 	}
 
-	@SideOnly(Side.CLIENT)
-	@Override
-	public IIcon getIconFromDamage(int meta) {
-		return icons[meta >= icons.length ? 0 : meta];
-	}
+    @SideOnly(Side.CLIENT)
```
Comment (glowredman) [inline src/main/java/baubles/common/ItemDebugger.java:36]: `@SideOnly(Side.CLIENT)` is not needed here

File: src/main/java/baubles/common/ItemDebugger.java [hunk 4]
```diff
@@ -38,11 +33,11 @@ public void registerIcons(IIconRegister ir) {
...
+    @Override
+    public IIcon getIconFromDamage(int meta) {
+        return icons[meta >= icons.length ? 0 : meta];
+    }
 
     @SuppressWarnings({"unchecked", "rawtypes"})
 	@SideOnly(Side.CLIENT)
```
Comment (glowredman) [inline src/main/java/baubles/common/ItemDebugger.java:43]: `@SideOnly(Side.CLIENT)` is not needed here

File: src/main/java/baubles/common/ItemDebugger.java [hunk 5]
```diff
@@ -75,40 +65,6 @@ public BaubleType getBaubleType(ItemStack itemStack) {
...
-	public boolean canEquip(ItemStack itemStack, EntityLivingBase player) {
-	  return true;
-	}
-
-	@Override
-	public boolean canUnequip(ItemStack itemStack, EntityLivingBase player) {
-	  return true;
-	}
-
 	@SideOnly(Side.CLIENT)
```
Comment (glowredman) [inline src/main/java/baubles/common/ItemDebugger.java:68]: `@SideOnly(Side.CLIENT)` is not needed here

**Comment by Caedis** [comment]
@glowredman why are they not needed? Is it because the server would never call them anyway?

**Comment by glowredman** [comment]
> @glowredman why are they not needed? Is it because the server would never call them anyway?

These methods/fields don't have any client-only types in their signatures (`IIcon` is not in `net.minecraft.client.**`, even though one would assume so).
That they aren't called is also true but doesn't matter, the server would still crash if e.g. `Item.registerIcons()` *didn't* have an `@SideOnly(Side.CLIENT)` annotation (`IIconRegister` *is* in `net.minecraft.client.**` and consequently marked as client-only).

**Comment by Caedis** [comment]
Good to know, thanks

**Pull request merged**